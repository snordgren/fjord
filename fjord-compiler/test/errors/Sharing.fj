module codegen.expressions.Sharing

import TestUtil

useSharedInt : Int -> Int -> Int
useSharedInt x y =
  rm x y

shareInt : Int -> Int
shareInt x =
  useSharedInt x x

myInt : Int
myInt = shareInt 5
