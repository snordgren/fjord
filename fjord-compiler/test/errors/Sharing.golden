test/errors/Sharing.fj:11:18:
   |
11 |   useSharedInt x x
   |                  ^^
too many usages of x
